fedimint-api/src/db/mod.rs

```diff
@@ -205,6 +300,26 @@ impl<'a> DatabaseTransaction<'a> {
         }
     }
 
+    pub fn with_module_prefix<'isolated>(
+        &'isolated mut self,
+        module_instance_id: ModuleInstanceId,
+    ) -> DatabaseTransaction<'isolated>
+    where
+        'parent: 'isolated,
+    {
+        let decoders = self.decoders.clone();
+        let isolated = Box::new(IsolatedDatabaseTransaction::new(self, module_instance_id));
+        DatabaseTransaction {
+            tx: isolated,
+            decoders,
+            // DatabaseTransaction passed to modules cannot be committed, so the commit tracker is set to committed to surpress the warning
```

Comment:
The parent tx will take care of it though, right?

Reply:
Yes, parent transaction will keep track of writes and the commit tracker. I can update the comment to reflect